build-tools/Xamarin.Android.Tools.BootstrapTasks/Xamarin.Android.Tools.BootstrapTasks/CreateAndroidEmulator.cs

```diff
@@ -1,190 +1,155 @@
 		{
-			if (string.IsNullOrEmpty (TargetId) && !string.IsNullOrEmpty (SdkVersion)) {
-				TargetId    = "system-images;android-" + SdkVersion + ";default;" + AndroidAbi;
-			}
-			Log.LogMessage (MessageImportance.Low, $"Task {nameof (CreateAndroidEmulator)}");
-			Log.LogMessage (MessageImportance.Low, $"  {nameof (AndroidAbi)}: {AndroidAbi}");
-			Log.LogMessage (MessageImportance.Low, $"  {nameof (AndroidSdkHome)}: {AndroidSdkHome}");
-			Log.LogMessage (MessageImportance.Low, $"  {nameof (JavaSdkHome)}: {JavaSdkHome}");
-			Log.LogMessage (MessageImportance.Low, $"  {nameof (ImageName)}: {ImageName}");
-			Log.LogMessage (MessageImportance.Low, $"  {nameof (SdkVersion)}: {SdkVersion}");
-			Log.LogMessage (MessageImportance.Low, $"  {nameof (TargetId)}: {TargetId}");
-			Log.LogMessage (MessageImportance.Low, $"  {nameof (ToolExe)}: {ToolExe}");
-			Log.LogMessage (MessageImportance.Low, $"  {nameof (ToolPath)}: {ToolPath}");
-
-			Run (GetAndroidPath ());
-
-			return !Log.HasLoggedErrors;
-		}
-
-		string GetAndroidPath ()
-		{
-			if (string.IsNullOrEmpty (ToolExe))
-				ToolExe = "avdmanager";
-
 			var dirs = string.IsNullOrEmpty (ToolPath)
 				? null
 				: new [] { ToolPath };
-			string filename;
-			var path = Which.GetProgramLocation (ToolExe, out filename, dirs);
+			var path = Which.GetProgramLocation (ToolExe ?? ToolName, out var filename, dirs);
 			if (path == null) {
 				Log.LogError ($"Could not find `avdmanager`. Please set the `{nameof (CreateAndroidEmulator)}.{nameof (ToolPath)}` property appropriately.");
-				return null;
+				return false;
 			}
-			return path;
-		}
+			ToolExe = filename;
 
-		void Run (string android)
-		{
-			if (android == null)
-				return;
+			if (string.IsNullOrEmpty (TargetId) && !string.IsNullOrEmpty (SdkVersion)) {
+				TargetId = "system-images;android-" + SdkVersion + ";default;" + AndroidAbi;
+			}
 
-			var arguments   = $"create avd --abi {AndroidAbi} -f -n {ImageName} --package \"{TargetId}\"";
-			Exec (android, arguments);
+			base.Execute ();
 
-			string configPath = Path.Combine (Environment.GetFolderPath (Environment.SpecialFolder.Personal), ".android", "avd", $"{ImageName}.avd", "config.ini");
-			if (!File.Exists (configPath)) {
-				Log.LogWarning ($"Config file for AVD '{ImageName}' not found at {configPath}");
-				Log.LogWarning ($"AVD '{ImageName}' will use default emulator settings (memory and data partition size)");
-				return;
-			}
+			if (Log.HasLoggedErrors)
+				return false;
 
-			ulong diskSize;
-			if (!UInt64.TryParse (DataPartitionSizeMB, out diskSize))
-				Log.LogError ($"Invalid data partition size '{DataPartitionSizeMB}' - must be a positive integer value expressing size in megabytes");
+			var configPaths = new [] {
+				Path.Combine (Environment.GetFolderPath (Environment.SpecialFolder.Personal), ".android", "avd", $"{ImageName}.avd", "config.ini"),
+				Path.Combine (AndroidSdkHome, ".android", "avd", $"{ImageName}.avd", "config.ini")
+			};
```

Comment:
I found the first path worked on macOS, but on Windows the file is in `%USERPROFILE%\android-toolchain\sdk\.android\`. Not sure why the path is different?